packages/web3-eth-abi/src/decode_contract_error_data.ts

```diff
@@ -0,0 +1,50 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { Eip838ExecutionError } from 'web3-errors';
+import { AbiErrorFragment } from 'web3-types';
+
+import { encodeErrorSignature } from './api/errors_api';
+import { decodeParameters } from './api/parameters_api';
+import { jsonInterfaceMethodToString } from './utils';
+
+export const decodeContractErrorData = (
```

Comment:
This method was moved from `web3-eth-contract` package to `web3-eth-abi` to avoid a circular dependency issue between `web3-eth-contract` and `web3-eth` since it's now being used inside `web3-eth`'s `getRevertReason` util method. It was also renamed from `decodeErrorData`